lib/service/services_cli.rb

```diff
@@ -139,10 +139,13 @@ def stop(targets, verbose: false)
             quiet_system System.launchctl, "bootout", "#{System.domain_target}/#{service.service_name}"
           end
           quiet_system System.launchctl, "stop", "#{System.domain_target}/#{service.service_name}" if service.pid?
-          rm service.dest if service.dest.exist?
         end
 
-        if service.pid?
+        rm service.dest if service.dest.exist?
```

Comment:
Currently the behavior of the `brew services stop` command is different on Linux and MacOS.

MacOS: Stops, disables, and unloads the service (plus deletes the service file).
Linux: Stops and disables the service.

This change aims to make `brew services stop` behave the same way on both platforms. We could always decide that that's not how we want that command to work though and we're okay with the Linux and MacOS versions working differently but then that should probably be documented somewhere.

Reply:
I'm fine with this either way. I think consistency is good but if there's reasons @SMillerDev can provide that doesn't make sense to provide it here: fine with me.

Reply:
If I remember correctly, macOS does not have a way to "disable" a service like systemd has. That's why we remove files, otherwise it would restart when the user logs in again. If that isn't the case I think keeping the files installed is fine.

Reply:
I think you're right about that limitation on macOS. Newer versions of launchctl have a disable command but I'm not sure how backwards compatible that is or what versions of macOS brew services even supports. Do either of you have any idea when that changed? If that's an option, I think it would be trivial to just change the stop command to never delete service files for both launchctl and systemctl.

If we assume that's not an option, there are still the following problems with the current implementation.

1. The FormulaWrapper#loaded? method always returns true on Linux.
2. This means that parts of the ServicesCli#stop method are unreachable on Linux.
3. The ServicesCli#run method fails if a systemd service file is not already installed.
4. `brew services list` returns different results on macOS and Linux after calling `brew services stop` making testing a tiny bit more complicated.

All of those problems can be resolved in other ways but I think the refactor would be slightly more complex and harder to maintain than what I've proposed here. Off the top of my head we could also do the following.

1. Split the ServicesCli#stop method into #launchctl_stop and #systemctl_stop.
2. Create a FormulaWrapper#disabled? method used in ServicesCli#systemctl_stop.
3. Test for none or stopped in the functional tests.

I guess the thing is removing the service file on Linux is probably functionally unnecessary but also creates consistency and I don't think there's any real downside to it (but I could be wrong about that).

Reply:
> I guess the thing is removing the service file on Linux is probably functionally unnecessary but also creates consistency and I don't think there's any real downside to it (but I could be wrong about that).

This reasoning makes sense to me and is fine with me if it is with @SMillerDev.

Reply:
I think it's fine too. I later realized that without removal it would be very inconvenient once you uninstall something you stopped.